stats/stats/src/charts/lines/new_contracts.rs

```diff
@@ -16,45 +20,45 @@ impl ChartPartialUpdater for NewContracts {
+        let query_maker = |from_: NaiveDate, to_: NaiveDate| {
+            Statement::from_sql_and_values(
                 DbBackend::Postgres,
-                r#"SELECT
-                    DATE(b."timestamp") as date,
-                    COUNT(*)::TEXT as value
-                FROM addresses a
-                    LEFT JOIN transactions t ON a.hash = t.created_contract_address_hash
-                    LEFT JOIN internal_transactions it ON a.hash = it.created_contract_address_hash
-                    JOIN blocks b ON b.hash = it.block_hash OR b.hash = t.block_hash
-                WHERE
-                    a.contract_code NOTNULL AND
-                    b.consensus = true AND
-                    DATE(b.timestamp) > $1
-                GROUP BY DATE(b."timestamp")"#,
-                vec![row.date.into()],
-            ),
-            None => Statement::from_sql_and_values(
-                DbBackend::Postgres,
-                r#"SELECT
-                    DATE(b."timestamp") as date,
-                    COUNT(*)::TEXT as value
-                FROM addresses a
-                    LEFT JOIN transactions t ON a.hash = t.created_contract_address_hash
-                    LEFT JOIN internal_transactions it ON a.hash = it.created_contract_address_hash
-                    JOIN blocks b ON b.hash = it.block_hash OR b.hash = t.block_hash
-                WHERE
-                    a.contract_code NOTNULL AND
-                    b.consensus = true
-                GROUP BY DATE(b."timestamp")"#,
-                vec![],
-            ),
+                r#"SELECT day AS date, COUNT(*)::text AS value
+                FROM (
+                    SELECT 
+                        DISTINCT ON (txns_plus_internal_txns.hash)
+                        txns_plus_internal_txns.day
+                    FROM (
+                        SELECT
+                            t.created_contract_address_hash AS hash,
+                            b.timestamp::date AS day
+                        FROM transactions t
+                            JOIN blocks b ON b.hash = t.block_hash
+                        WHERE
+                            t.created_contract_address_hash NOTNULL AND
+                            b.consensus = TRUE AND
+                            b.timestamp::date < $2 AND
+                            b.timestamp::date >= $1
+                        UNION
+                        SELECT
+                            it.created_contract_address_hash AS hash,
+                            b.timestamp::date AS day
+                        FROM internal_transactions it
+                            JOIN blocks b ON b.hash = it.block_hash
+                        WHERE
+                            it.created_contract_address_hash NOTNULL AND
+                            b.consensus = TRUE AND
+                            b.timestamp::date < $2 AND
+                            b.timestamp::date >= $1
+                    ) txns_plus_internal_txns
```

Comment:
Probably also need to order by day, as `DISTINCT ON` would use a random order otherwise (forgot to write about that above)

https://www.postgresql.org/docs/current/sql-select.html#SQL-DISTINCT